pulsar-client-cpp/lib/ConsumerImpl.h

```diff
@@ -135,6 +137,7 @@ enum ConsumerTopicType {
     ExecutorServicePtr listenerExecutor_;
     ConsumerTopicType consumerTopicType_;
     UnboundedBlockingQueue<Message> incomingMessages_;
+    UnboundedBlockingQueue<ReceiveCallback> pendingReceives_;
```

Comment:
UnboundedBlockingQueue is not a good idea, since you don't want to block if the queue is empty
I would suggest using a vector or a list.